src/ekat/CMakeLists.txt

```diff
@@ -129,7 +129,7 @@ install (DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ekat_f90_modules
 add_library(ekat_test_main util/ekat_catch_main.cpp)
 target_link_libraries(ekat_test_main PUBLIC ekat)
 target_include_directories(ekat_test_main PUBLIC
-  $<BUILD_INTERFACE:${EKAT_SOURCE_DIR}/extern/catch2/include>
+  $<BUILD_INTERFACE:${EKAT_SOURCE_DIR}/extern/Catch2/src>
```

Comment:
Do we need this? That folder is already in the PUBLIC section of the catch 2 lib.

Reply:
Actually, I just noticed that we do not link against the Catch2 target. Any reason not to? I like to leverage cmake's targets property propagation...

Reply:
Don't we? It's included in `EKAT_TPL_LIBRARIES`.

Reply:
Ah, right. I didn't see it in the line above, but since it's linked to ekat, it's linked to ekat_catch_main as well.

So my first comment stands: we can remove the `target_include_directories` line (one less generator expression to mentally parse!)